src/API/Instrumentation/AutoInstrumentation/HookManagerInterface.php

```diff
@@ -9,10 +9,9 @@
 
 interface HookManagerInterface
 {
-
     /**
      * @param Closure(object|string|null,array,string,string,string|null,int|null):void|null $preHook
-     * @param Closure(object|string|null,array,mixed,Throwable|null,string,string,string|null,int|null):void|null $postHook
+     * @param Closure(object|string|null,array,mixed,Throwable|null):void|null $postHook
```

Comment:
Definitely those extra args in post are wrong, but it looks wonky in other ways too. I _think_ this is correct?
```
     * @param ?Closure(object|string|null,array,?string,string,?string,?int):array $preHook
     * @param ?Closure(object|string|null,array,mixed,?Throwable):mixed $postHook
```

Reply:
Yeah, thanks for following up with that. I put the PR back into draft as I thought return type also looked wonky.

I shall take a further look and see if I can get phan to be happy with the changes 👍

Reply:
@brettmc I've updated the PR with those changes.

I was still getting complaints with the preHook expecting an array, where it actually appears to permit void|null too. Marking the return of pre as `?array` suppressed the phan complaints for me - so hoping this is okay? :thinking:

Reply:
(Fixing the build first, oops!)

Reply:
Phan seemed a bit precious around parsing a mixed return type of `@param` - but seems happier with wrapping it in parentheses.

I have seen pre-hooks without an explicit return (thus void) which I thought would might benefit to add?

Reply:
Yes you are right, it's definitely allowed to return nothing from hooks.